tests/shippers/test_logstash.py

```diff
@@ -0,0 +1,193 @@
+                    "name": "arn:aws:s3:::bucket_name",
+                    "arn": "bucket_name",
+                },
+                "object": {
+                    "key": "file.key",
+                },
+            },
+        },
+        "cloud": {
+            "provider": "aws",
+            "region": "aws-region",
+        },
+    },
+    "meta": {},
+}
+
+_username = "admin"
+_password = "password"
+
+
+def _dummy_replay_handler(output_type: str, output_args: dict[str, Any], event_payload: dict[str, Any]) -> None:
+    assert output_type == "logstash"
+    assert event_payload == _dummy_event
+
+
+@pytest.mark.unit
+class TestLogstashShipper(TestCase):
+    @responses.activate
+    def test_send_successful(self) -> None:
+        def request_callback(request: PreparedRequest) -> tuple[int, dict[Any, Any], str]:
+            _payload = []
+            assert request.headers["Content-Encoding"] == "gzip"
+            assert request.headers["Content-Type"] == "application/x-ndjson"
+            assert request.body is not None
+            assert isinstance(request.body, bytes)
+
+            events = gzip.decompress(request.body).decode("utf-8").split("\n")
+            for event in events:
+                _payload.append(ujson.loads(event))
+
+            expected_event = _dummy_event
+            expected_event["_id"] = "_id"
+            assert _payload == [expected_event, expected_event]
+
+            return 200, {}, "okay"
+
+        def event_id_generator(event: dict[str, Any]) -> str:
+            return "_id"
+
+        url = "http://logstash_url"
+        responses.add_callback(responses.PUT, url, callback=request_callback)
+        logstash_shipper = LogstashShipper(logstash_url=url, max_batch_size=2)
+        logstash_shipper.set_event_id_generator(event_id_generator)
+        logstash_shipper.send(_dummy_event)
+        logstash_shipper.send(_dummy_event)
+
+    @responses.activate
+    def test_send_failures(self) -> None:
+        url = "http://logstash_url"
+        with self.subTest("Does not exceed max_retries"):
+            responses.put(url=url, status=429)
+            responses.put(url=url, status=429)
+            responses.put(url=url, status=429)
+            responses.put(url=url, status=200)
+            logstash_shipper = LogstashShipper(logstash_url=url)
+            assert logstash_shipper.send(_dummy_event) == _EVENT_SENT
+        with self.subTest("Exceeds max retries, replay handler set"):
+            for i in range(_MAX_RETRIES):
+                responses.put(url=url, status=429)
+            responses.put(url=url, status=429)
+            logstash_shipper = LogstashShipper(logstash_url=url)
+            replay_handler = MagicMock(side_effect=_dummy_replay_handler)
+            logstash_shipper.set_replay_handler(replay_handler)
+            assert logstash_shipper.send(_dummy_event) == _EVENT_SENT
+            replay_handler.assert_called_once_with("logstash", {}, _dummy_event)
```

Comment:
```
replay_handler.assert_called_once_with("logstash", {}, _dummy_event)
```
the second argument should contain the `output_args`, in this case i'd expect `logstash_url`

Reply:
`output_args` are not really used for Logstash, but I added it anyway as it can be helpful for users to troubleshoot errors ending up in the replay queue

Reply:
> `output_args` are not really used for Logstash

where the original output args like compression level etc come from if they are not in the `output_args` of the replay queue message?